storage/database/src/main/java/de/danoeh/antennapod/storage/database/ReleaseScheduleGuesser.java

```diff
@@ -122,13 +126,81 @@ private static Stats getStats(List<Date> releaseDates) {
                 daysOfWeek, daysOfMonth, mostOftenDayOfWeek, mostOftenDayOfMonth);
     }
 
+    private static List<Integer> getLargeDays(ReleaseScheduleGuesser.Stats stats, int maxDaysOff) {
+        List<Integer> largeDays = new ArrayList<>();
+        for (int i = Calendar.SUNDAY; i <= Calendar.SATURDAY; i++) {
+            if (stats.daysOfWeek[i] > maxDaysOff) {
+                largeDays.add(i);
+            }
+        }
+        return largeDays;
+    }
+
+    private static Date getUniqueDate(Date date) {
+        GregorianCalendar current = new GregorianCalendar();
+        current.setTime(date);
+        current.set(Calendar.HOUR_OF_DAY, 0);
+        current.set(Calendar.MINUTE, 0);
+        current.set(Calendar.SECOND, 0);
+        current.set(Calendar.MILLISECOND, 0);
+        return current.getTime();
+    }
+
+    private static List<Date> getMostRecentDates(List<Date> releaseDates, int amount) {
+        Set<String> uniqueDates = new HashSet<>();
+        int i = releaseDates.size();
+        while (i > 0 && amount > 0) {
+            i--;
+            String dateString = getUniqueDate(releaseDates.get(i)).toString();
+            if (!uniqueDates.contains(dateString)) {
+                uniqueDates.add(dateString);
+                amount--;
+            }
+        }
+        return releaseDates.subList(i, releaseDates.size());
+    }
+
+    private static boolean containsMultipleReleasesPerDay(List<Date> releaseDates, int threshold) {
```

Comment:
This is only called once, with a specific threshold. Please remove the threshold and move it into this function. This makes it easier to read for future developers.